lib/pavilion/expression_functions/base.py

```diff
@@ -36,6 +36,43 @@ def num(val):
 
     raise RuntimeError("Invalid value '{}' given to num.".format(val))
 
+def flag(val: str, flag_str: str) -> str:
+    """Return a flag string for a boolean variable, if
+    the variable is set, or an empty string otherwise."""
+
+    if val.lower() in {'true', 'on', 'yes'}:
+        return flag_str
+    elif val.lower() in {'false', 'off', 'no'}:
+        return ''
+    else:
+        raise ValueError(f'Could not convert {val} into boolean-like.')
```

Comment:
Reverse the argument order. 

`flag('-some_opt', some_opt_var)` reads a lot cleaner to me.